### ISSUE TITLE
backgroundImage not displayed in Timeline

### ISSUE THREAD

**Issue by da2x** [DominicMaas/TimelineExtension#23]
Pages with og:image doesn’t show image in Timeline. Not sure what adaptive card is being used by Edge. Looked over the documentation but still not sure what is missing.

**Comment by DominicMaas** [comment]
It was working before. Did this pull request maybe break it?

https://github.com/DominicMaas/TimelineExtension/pull/14

**Comment by DominicMaas** [comment]
Huh, I had it working on both Chrome and Firefox six or so hours ago.

**Comment by da2x** [comment]
I just tested Vivaldi, Chrome, and Firefo’. None of them load background images for any of these despite background images being detected correctly:
* https://www.neowin.net/news/latest-firefox-nightly-build-adds-support-for-windows-10-dark-theme
* https://www.ctrl.blog/entry/firefox-u2f-google
* https://www.keycdn.com/blog/pop-in-melbourne-perth/

**Comment by 810** [comment]
I see only active activity, that one is with no background. Just like Edge.

I think the activity isn't saved, when I go to earlier today- see all activity .  Then I see only edge and Phpstorm

**Comment by DominicMaas** [comment]
I assume that you have logged into your Microsoft Account (in the extension)?

**Comment by 810** [comment]
Yes, the project rome is working perfect. Only I can't see the google activity webpages on "See all activity"

On the Microsoft privacy activity I can't see it. 
![test](https://user-images.githubusercontent.com/876623/42728599-63d06934-87ad-11e8-9eef-c09e84f91732.jpg)

**Comment by DominicMaas** [comment]
Hmmm. Alright, could you go to the Chome extension page (chrome://extensions). Under this extension (Windows Timeline Support) there is a link that says "background page". Could you please click on it, navigate to the console and take a screenshot (or copy paste any errors that are appearing).

**Comment by 810** [comment]
![test](https://user-images.githubusercontent.com/876623/42728649-97df3e70-87ae-11e8-8503-525590f75dd2.jpg)

**Comment by DominicMaas** [comment]
Does nothing show when you navigate to other pages? (Try searching for something on Google, keep in mind you have to wait ~8 seconds after page load before the extension sends an activity.

**Comment by 810** [comment]
yes, nothing is stored.

**Comment by DominicMaas** [comment]
On the "Default Levels" drop down box (Dev Tools), can you tick "Verbose" and see if anything is displayed?

If not, try removing the extension and adding it again (as a potential temp fix), I'll add some more logging in the mean time.

**Comment by 810** [comment]
ok on Stable I get this error now:

```
graph.microsoft.com/v1.0/me/activities/https%3A__login.live.com_ppsecure_post.srf%3Fwa%3Dwsignin1.0%26rpsnv%3D13%26ct%3D1531605334%26rver%3D6.7.6643.0%26wp%3DMBI_SSL%26wreply%3Dhttps%3A%252F%252Faccount.microsoft.com%252Fauth%252Fcomplete-signin%253Fru%253Dhttps%25253A%25252F%25252Faccount.microsoft.com%25252Fprivacy%25253Fru%25253Dhttps%2525253A%2525252F%2525252Faccount.microsoft.com%2525252Fprivacy%2525252Factivity-history%26lc%3D1043%26id%3D292666%26lw%3D1%26fl%3Deasi2%26contextid%3D75C98387C6B3EC9A%26bk%3D1531605337%26uaid%3Dc75ceaebb9c345ccb9c3a784357b5219%26pid%3D0:1 Failed to load resource: the server responded with a status of 400 (Bad Request)
````

**Comment by 810** [comment]
![test](https://user-images.githubusercontent.com/876623/42728778-5785f6da-87b2-11e8-8279-2e3f2a7d25e8.jpg)

**Comment by 810** [comment]
![test1](https://user-images.githubusercontent.com/876623/42728790-c6898754-87b2-11e8-9e1e-3e629e9cc65a.jpg)

**Comment by DominicMaas** [comment]
Sometimes those errors are produced (certain urls contains characters that break everything), but it looks like the other requests were successful, (https://www.ajax.nl/home and https://account.microsoft.com/privacy). Do these not appear in Windows Timeline?

Another thing I can think of is restarting explorer.exe on your PC (this also restarts timeline - messy, but it has worked for me in the past).

**Comment by 810** [comment]
No, I can't see them on the timeline or on https://account.microsoft.com/privacy/activity-history?view=browse

**Comment by DominicMaas** [comment]
Yeah that's usually the id that works (I'm going to push an update that base64 encodes the url anyway, to avoid the bad request errors). But it looks like everything posted successfully. 

Only thing I can think of is if the time zone, or created time is wrong (if it's in the future it may not show anywhere, but still exist).

**Comment by 810** [comment]
i have change it to let activityId = webActivity.activityOriginUrl;
reloaded the extension and reload chrome, and now i can see it on the activities

**Comment by DominicMaas** [comment]
Could you pull the latest commit and see if that also works?

**Comment by DominicMaas** [comment]
Wait, I made a mistake, just need to fix it quickly

**Comment by DominicMaas** [comment]
Okay, now try.

**Comment by 810** [comment]
Yes, its working now.  Only the time is set to utc.  So for us its missing 2 hours. (new day)

**Comment by 810** [comment]
Now on issue, i still can't see the og:image

**Comment by DominicMaas** [comment]
Could you navigate to https://github.com/DominicMaas/TimelineExtension, and copy the data payload (starts with `{"appActivityId"`... - is written into the console) and paste it here?

**Comment by 810** [comment]
```
"activitySourceHost":"github.com","createdDateTime":"2018-07-14T22:51:10.243Z","lastModifiedDateTime":"2018-07-14T22:51:10.555Z","id":"5044551257986403992","appActivityId":"aHR0cHM6Ly9naXRodWIuY29tL0RvbWluaWNNYWFzL1RpbWVsaW5lRXh0ZW5zaW9u","activationUrl":"https://github.com/DominicMaas/TimelineExtension","appDisplayName":"Google Chrome","userTimezone":"Europe/Berlin","fallbackUrl":"https://github.com/DominicMaas/TimelineExtension","expirationDateTime":"2018-08-13T22:51:10.243Z","status":"active","visualElements":{"displayText":"DominicMaas/TimelineExtension","description":"https://github.com/DominicMaas/TimelineExtension","content":{"$schema":"http://adaptivecards.io/schemas/adaptive-card.json","type":"AdaptiveCard","backgroundImage":"https://avatars3.githubusercontent.com/u/5589453?s=400&v=4","body":[{"type":"Container","items":[{"type":"TextBlock","text":"DominicMaas/TimelineExtension","weight":"bolder","size":"large","wrap":true,"maxLines":3},{"type":"TextBlock","text":"https://github.com/DominicMaas/TimelineExtension","size":"default","wrap":true,"maxLines":3}]}]},"attribution":{"iconUrl":"https://assets-cdn.github.com/favicon.ico","alternateText":"github.com","addImageQuery":false}},"historyItems@odata.context":"https://graph.microsoft.com/v1.0/$metadata#users('schumacher1986%40hotmail.com')/activities('5044551257986403992')/historyItems","historyItems":[{"status":"updated","userTimezone":"Europe/Berlin","createdDateTime":"2018-07-14T22:51:10.243Z","lastModifiedDateTime":"2018-07-14T22:51:10.243Z","id":"e071f9c6-6e20-4b51-9730-61d465ee8b69","startedDateTime":"2018-07-14T22:51:10.042Z","lastActiveDateTime":"2018-07-14T22:51:10.042Z","expirationDateTime":"2018-08-13T22:51:10.243Z","activeDurationSeconds":0}]}

```

**Comment by DominicMaas** [comment]
No background image in the Timeline card? Can you access the following url and see the image? 

https://avatars3.githubusercontent.com/u/5589453?s=400&v=4

**Comment by 810** [comment]
its now fluent, maybe its an api issue .

I can see the image, and the data payload looks ok for me.

**Comment by DominicMaas** [comment]
@da2x Are you still having this issue?

**Comment by 810** [comment]
its still there, on the menu activity, i see it correct, but on the real timeline I see the fluent.

**Comment by DominicMaas** [comment]
Could maybe be a Timeline bug? Are you running insider preview?

**Comment by 810** [comment]
can be yes, im on 17713

**Comment by dwm1909** [comment]
Am also see it like fluent into real timeline with insiders build 17735.1000

**Comment by 810** [comment]
please upvote: https://aka.ms/AA220f7

**Comment by DominicMaas** [comment]
Does this only happen on preview versions? Can you see the background on the stable version of Windows 10?

**Comment by dwm1909** [comment]
same issues even with 1803 17134.191 and that applays as well for Firefox.

**Comment by DominicMaas** [comment]
Strange, no matter what I do, I cannot reproduce the error. So no activities generated by this extension have a background at all? Can you see the background under the "Recent Activities" menu in the extension?

**Comment by dwm1909** [comment]
Non activits have any background only them frome Edge, not frome firefox/Chrome but for me its dosent do so mutch dont relly have the activities for watch the background, best i would think is to upvote the issues on the feedback hub as 810 with the AKA link see if microsoft can fix it by there ends.

**Comment by 810** [comment]
![image](https://user-images.githubusercontent.com/876623/44052980-970b2688-9f3e-11e8-8507-99b878951d8a.png)

**Comment by dwm1909** [comment]
Yes insade of the Extention working but not in the real timeline but not all pages has a backrund in the extention as well but most of them.

**Comment by fuzuyx** [comment]
any updates/workaround?
